pkg/resources/kafka/configmap.go

```diff
@@ -63,49 +63,36 @@ func (r *Reconciler) getConfigProperties(bConfig *v1beta1.BrokerConfig, id int32
 		log.Error(err, "setting zookeeper.connect parameter in broker configuration resulted an error")
 	}
 
-	// Add SSL configuration
-	if r.KafkaCluster.Spec.ListenersConfig.SSLSecrets != nil {
-		if err := config.Set("ssl.keystore.location", serverKeystorePath+"/"+v1alpha1.TLSJKSKeyStore); err != nil {
-			log.Error(err, "setting ssl.keystore.location parameter in broker configuration resulted an error")
+	brokerPort, err := kafkautils.GetBrokerContainerPort(r.KafkaCluster)
+	if err != nil {
+		log.Error(err, "getting Kafka broker port for Cruise Control failed")
+	}
+	// Add Cruise Control SSL configuration
+	if util.IsSSLEnabledForInternalCommunication(r.KafkaCluster.Spec.ListenersConfig.InternalListeners) {
+		if !r.KafkaCluster.Spec.IsClientSSLSecretPresent() {
+			log.Error(errors.New("cruise control metrics reporter needs ssl but client certificate hasn't specified"), "")
 		}
-		if err := config.Set("ssl.truststore.location", serverKeystorePath+"/"+v1alpha1.TLSJKSTrustStore); err != nil {
-			log.Error(err, "setting ssl.truststore.location parameter in broker configuration resulted an error")
+		if err := config.Set("cruise.control.metrics.reporter.security.protocol", "SSL"); err != nil {
```

Comment:
The same approach could be used as in https://github.com/banzaicloud/koperator/pull/698/files#diff-712b068c8dcc08d66a78aa61aedf8108f247eb9d3a6ee68d3d44047d1dfc5fe4R251

Reply:
Yes, It can be replaced some places in the codebase. I think this is out of the scope.